libs/platform/ebpf_ring_buffer.h

```diff
@@ -0,0 +1,130 @@
+        size_t reserved_epoch;
+        uint8_t data[1];
+    } ebpf_ring_buffer_record_t;
+
+    typedef struct _ebpf_ring_buffer ebpf_ring_buffer_t;
+
+    /**
+     * @brief Allocate a ring_buffer with capacity.
+     *
+     * @param[out] ring_buffer Pointer to buffer that holds ringbuffer pointer on success.
+     * @param[in] capacity Size in bytes of ring buffer.
+     * @retval EPBF_SUCCESS Successfully allocated ringbuffer.
+     * @retval EBPF_NO_MEMORY Unable to allocate ringbuffer.
+     */
+    ebpf_result_t
+    ebpf_ring_buffer_create(_Outptr_ ebpf_ring_buffer_t** ring_buffer, size_t capacity);
+
+    /**
+     * @brief Free a ring buffer.
+     *
+     * @param[in] ring_buffer Ringbuffer to free.
+     */
+    void
+    ebpf_ring_buffer_destroy(_Frees_ptr_opt_ ebpf_ring_buffer_t* ring_buffer);
+
+    /**
+     * @brief Write out a variable sized record to the ringbuffer.
+     *
+     * @param[in] ring_buffer Ringbuffer to write to.
+     * @param[in] data Data to copy into record.
+     * @param[in] length Length of data to copy
+     * @retval EPBF_SUCCESS Successfully wrote record ringbuffer.
+     * @retval EBPF_INVALID_ARGUMENT Unable to output to ring buffer.
+     */
+    ebpf_result_t
+    ebpf_ring_buffer_output(
+        _In_ ebpf_ring_buffer_t* ring_buffer, _In_reads_bytes_(length) uint8_t* data, size_t length);
+
+    /**
+     * @brief Query the current ready and free offsets from the ringbuffer.
+     *
+     * @param[in] ring_buffer Ringbuffer to query.
+     * @param[in] flags Flags to control query behavior.
+     * @param[in] consumer Offset of the first buffer that can be consumed.
+     * @param[in] producer Offset of the next buffer to be produced.
+     * @retval EPBF_SUCCESS Successfully queried ringbuffer.
+     */
+    ebpf_result_t
+    ebpf_ring_buffer_query(
+        _In_ ebpf_ring_buffer_t* ring_buffer, uint64_t flags, _Out_ size_t* consumer, _Out_ size_t* producer);
+
+    /**
+     * @brief Mark one or more records in the ring buffer as free.
+     *
+     * @param ring_buffer Ringbuffer to update.
+     * @param count Count of bytes to free in the ringbuffer.
+     * @retval EPBF_SUCCESS Successfully free to ringbuffer.
+     * @retval EBPF_INVALID_ARGUMENT Unable to free to ring buffer.
+     */
+    ebpf_result_t
+    ebpf_ring_buffer_free(_In_ ebpf_ring_buffer_t* ring_buffer, size_t count);
+
+    /**
+     * @brief Get pointer to the ringbuffer shared data.
+     *
+     * @param[in] ring_buffer Ring buffer to query.
+     * @param[out] buffer Pointer to ring buffer data.
+     * @retval EPBF_SUCCESS Successfully mapped the ringbuffer.
+     * @retval EBPF_INVALID_ARGUMENT Unable to map the ring buffer.
+     */
+    ebpf_result_t
+    ebpf_ring_buffer_map_buffer(_In_ ebpf_ring_buffer_t* ring_buffer, _Outptr_ uint8_t** buffer);
+
+    /**
+     * @brief Reserve a buffer in the ringbuffer. Buffer is valid until either ebpf_ring_buffer_commit,
```

Comment:
should we use "submit" instead of "commit" in the api name, for consistency with `bpf_ringbuf_submit()`?